h2/src/main/org/h2/engine/SysProperties.java

```diff
@@ -346,6 +346,15 @@
             Utils.getProperty("h2.oldStyleOuterJoin",
                     Constants.VERSION_MINOR < 4);
 
+    /**
+     * System property {@code h2.oldResultSetGetObject}, {@code false} by default.
+     * Return {@code Byte} and {@code Short} instead of {@code Integer} from
+     * {@code ResultSet#getObject(...)} for {@code TINYINT} and {@code SMALLINT}
+     * values.
+     */
+    public static final boolean OLD_RESULT_SET_GET_OBJECT =
+            Utils.getProperty("h2.oldResultSetGetObject", false);
+
```

Comment:
hmmm, we don't normally change API in a minor point release, so this needs to default to true for now.
At some point, we need to bump the version and then set all of these compat flag things to the newer stuff.